src/FormFileInput.js

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import FormContext from './FormContext';
+import { useBootstrapPrefix } from './ThemeProvider';
+
+const propTypes = {
+  /**
+   * @default 'form-file-input'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * A seperate bsPrefix used for custom controls
+   *
+   * @default 'custom-file-input'
+   */
+  bsCustomPrefix: PropTypes.string,
+
+  /**
+   * The underlying HTML element to use when rendering the FormFileInput.
+   *
+   * @type {('input'|elementType)}
+   */
+  as: PropTypes.elementType,
+
+  /** A HTML id attribute, necessary for proper form accessibility. */
+  id: PropTypes.string,
```

Comment:
usually we have a requiredForA11y proptype for this

Reply:
You are correct. Looking into this now and I think we should do like this:

- FormFile
  - Component
    -   id: isRequiredForA11y(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
  - Types
    - id: string | number;
- FormFileInput
  - Component
    -   id: isRequiredForA11y(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
  - Types
    - id: string | number;
  - FormFileLabel
    - do we do the same thing here for htmlFor?

There seems to be a mix of how ID is defined between components as well. Do we have a set bet practice? I see things such as:
```
  id: isRequiredForA11y(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
```
as well as
`id: requiredForA11y(PropTypes.string),`

Reply:
Looking at FormCheck/Input/Label again. There it seems as if you are allowed to omit id and if you do controlId will be set to FormContext. Id is not forced. Perhaps the type description should be updated instead of forced?

`const { controlId, custom } = useContext(FormContext);`
and then
```
<Component
{...props}
ref={ref}
id={id || controlId}
className={classNames(
	className,
	bsPrefix,
	isValid && 'is-valid',
	isInvalid && 'is-invalid',
	isStatic && 'position-static',
)}
/>
```

Reply:
@jquense I took another look again.

If, for any reason, the Id is not set the label will not work properly when custom is not used. So I guess we should always force the Id.

And I guess we should do it like this?
```
  id: isRequiredForA11y(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
```

Plus update the types

Reply:
this looks right to me... especially given that the examples often show the file inputs in form groups